bundle/edu.gemini.sp.vcs/src/main/scala/edu/gemini/sp/vcs/diff/ObsEditValidator.scala

```diff
@@ -0,0 +1,121 @@
+package edu.gemini.sp.vcs.diff
+
+import edu.gemini.shared.util.VersionComparison.{Same, Older, Newer, Conflicting}
+import edu.gemini.sp.vcs.diff.MergeCorrection._
+import edu.gemini.sp.vcs.diff.ObsEdit.{ObsDelete, ObsUpdate, ObsCreate}
+import edu.gemini.spModel.core.SPProgramID
+import edu.gemini.spModel.gemini.security._
+import edu.gemini.spModel.gemini.security.UserRolePrivileges._
+import edu.gemini.spModel.obs._
+import edu.gemini.spModel.obs.ObservationStatus._
+import edu.gemini.spModel.too._
+import edu.gemini.util.security.permission.{PiPermission, NgoPermission, StaffPermission}
+
+import java.security.Permission
+
+import scalaz._
+import Scalaz._
+
+// NOTE: extracted from (and will replace):
+// edu.gemini.util.security.policy.MergeValidator
+
+/**
+ * Evaluates whether a local observation edit should be considered legal.
+ */
+class ObsEditValidator(privs: UserRolePrivileges, too: TooType) {
```

Comment:
This will be done as it is currently done in the OT, via "conflict" notes.  They appear in the program tree icons and when you click on a node with a note there is a brief description. It's much like P2 warnings but you can also cycle through the conflicts and acknowledge them to make them go away and allow you to merge.